src/controllers/networks/networks.ts

```diff
@@ -68,69 +73,239 @@ export class NetworksController extends EventEmitter {
   }
 
   get networks(): Network[] {
-    if (!this.#networks) return predefinedNetworks
+    if (!Object.keys(this.#networks).length) return predefinedNetworks
 
     const uniqueNetworksByChainId = Object.values(this.#networks)
       .sort((a, b) => +b.predefined - +a.predefined) // first predefined
       .filter((item, index, self) => self.findIndex((i) => i.chainId === item.chainId) === index) // unique by chainId (predefined with priority)
     return uniqueNetworksByChainId.map((network) => {
       // eslint-disable-next-line no-param-reassign
-      network.features = getFeaturesByNetworkProperties({
-        isSAEnabled: network.isSAEnabled,
-        isOptimistic: network.isOptimistic ?? false,
-        rpcNoStateOverride: network.rpcNoStateOverride,
-        erc4337: network.erc4337,
-        areContractsDeployed: network.areContractsDeployed,
-        feeOptions: network.feeOptions,
-        platformId: network.platformId,
-        nativeAssetId: network.nativeAssetId,
-        flagged: network.flagged ?? false,
-        chainId: network.chainId,
-        hasSingleton: network.hasSingleton
-      })
+      network.features = getFeaturesByNetworkProperties(
+        {
+          isSAEnabled: network.isSAEnabled,
+          isOptimistic: network.isOptimistic ?? false,
+          rpcNoStateOverride: network.rpcNoStateOverride,
+          erc4337: network.erc4337,
+          areContractsDeployed: network.areContractsDeployed,
+          feeOptions: network.feeOptions,
+          platformId: network.platformId,
+          nativeAssetId: network.nativeAssetId,
+          flagged: network.flagged ?? false,
+          chainId: network.chainId,
+          hasSingleton: network.hasSingleton
+        },
+        network
+      )
       return network
     })
   }
 
+  /**
+   * Loads and synchronizes network configurations from storage and the relayer.
+   *
+   * This method performs the following steps:
+   * 1. Retrieves the latest network configurations from storage.
+   * 2. Converts the network structure from [key: NetworkId] to [key: chainId].
+   * 3. If no networks are found in storage, sets predefined networks and emits an update.
+   * 4. Merges the networks from the Relayer with the stored networks.
+   * 5. Ensures predefined networks are marked correctly and handles special cases (e.g., Odyssey network).
+   * 6. Sorts networks with predefined ones first, followed by custom networks, ordered by chainId.
+   * 7. Updates the networks in storage.
+   * 8. Asynchronously updates network features if needed.
+   *
+   * This method ensures that the application has the most up-to-date network configurations,
+   * handles migration of legacy data, and maintains consistency between stored and relayer-provided networks.
+   */
```

Comment:
Well documented 👏 , we discussed offline that it would be great if we have time to write unit tests that would cover all these scenarios.

Reply:
I added a test for the merging relayer logic, will add more later on.